ast/Helpers.h

```diff
@@ -546,16 +546,26 @@ class MK {
         return send->fun == core::Names::new_() && send->recv.isSelfReference();
     }
 
+    /*
+     * Is this an expression that refers to `::T`?
+     *
+     * This helper works on a purely syntactic level. For instance, this function determines if an expression is `T`,
+     * either with no scope or with the root scope (i.e. `::T`). This might not actually refer to the `T` that we define
+     * for users, but we don't know that information at the AST level.
+     */
```

Comment:
@trevor-stripe looks like we already had a helper extracted, I moved the documentation to this one.